esphome/components/ct_clamp/ct_clamp_sensor.cpp

```diff
@@ -9,20 +9,31 @@ namespace ct_clamp {
 static const char *const TAG = "ct_clamp";
 
 void CTClampSensor::setup() {
+  // Setup only starts the calibration phase, in loop() the actual sampling is happening.
   this->is_calibrating_offset_ = true;
+
+  // Request a high loop() execution interval during calibration phase.
   this->high_freq_.start();
   this->set_timeout("calibrate_offset", this->sample_duration_, [this]() {
     this->high_freq_.stop();
     this->is_calibrating_offset_ = false;
-    if (this->num_samples_ != 0) {
+
+    if (this->differential_sampling_) {
+      this->offset_ = 0.0f;
+    } else if (this->num_samples_ != 0) {
       this->offset_ = this->sample_sum_ / this->num_samples_;
     }
   });
+
+  // Set sampling values
+  this->num_samples_ = 0;
+  this->sample_sum_ = 0.0f;
 }
 
 void CTClampSensor::dump_config() {
   LOG_SENSOR("", "CT Clamp Sensor", this);
   ESP_LOGCONFIG(TAG, "  Sample Duration: %.2fs", this->sample_duration_ / 1e3f);
+  ESP_LOGCONFIG(TAG, "  Differential Sampling: %s", this->differential_sampling_ ? "true" : "false");
```

Comment:
Can use one of the helpers for this
```suggestion
  ESP_LOGCONFIG(TAG, "  Differential Sampling: %s", YESNO(this->differential_sampling_);
```
or
```suggestion
  ESP_LOGCONFIG(TAG, "  Differential Sampling: %s", TRUEFALSE(this->differential_sampling_);
```